sdk/dotnet/Apps/V1/Deployment.cs

```diff
@@ -0,0 +1,122 @@
+    ///    'NewReplicaSetAvailable'. For generation &amp;lt;= 1, this status field does not exist,
+    ///    because it doesn't do a rollout (i.e., it simply creates the Deployment and
+    ///    corresponding ReplicaSet), and therefore there is no rollout to mark as 'Progressing'.
+    /// 
+    /// If the Deployment has not reached a Ready state after 10 minutes, it will
+    /// time out and mark the resource update as Failed. You can override the default timeout value
+    /// by setting the 'customTimeouts' option on the resource.
+    /// </summary>
+    public partial class Deployment : Pulumi.CustomResource
+    {
+        /// <summary>
+        /// APIVersion defines the versioned schema of this representation of an object. Servers
+        /// should convert recognized schemas to the latest internal value, and may reject
+        /// unrecognized values. More info:
+        /// https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
+        /// </summary>
+        [Output("apiVersion")]
+        public Output<string> ApiVersion { get; private set; } = null!;
+
+        /// <summary>
+        /// Kind is a string value representing the REST resource this object represents. Servers
+        /// may infer this from the endpoint the client submits requests to. Cannot be updated. In
+        /// CamelCase. More info:
+        /// https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+        /// </summary>
+        [Output("kind")]
+        public Output<string> Kind { get; private set; } = null!;
+
+        /// <summary>
+        /// Standard object metadata.
+        /// </summary>
+        [Output("metadata")]
+        public Output<Types.Outputs.Meta.V1.ObjectMeta> Metadata { get; private set; } = null!;
+
+        /// <summary>
+        /// Specification of the desired behavior of the Deployment.
+        /// </summary>
+        [Output("spec")]
+        public Output<Types.Outputs.Apps.V1.DeploymentSpec> Spec { get; private set; } = null!;
+
+        /// <summary>
+        /// Most recently observed status of the Deployment.
+        /// </summary>
+        [Output("status")]
+        public Output<Types.Outputs.Apps.V1.DeploymentStatus> Status { get; private set; } = null!;
+
+
+        /// <summary>
+        /// Create a Deployment resource with the given unique name, arguments, and options.
+        /// </summary>
+        /// <param name="name">The unique name of the resource</param>
+        /// <param name="args">The arguments used to populate this resource's properties</param>
+        /// <param name="options">A bag of options that control this resource's behavior</param>
+        public Deployment(string name, Types.Inputs.Apps.V1.DeploymentArgs? args = null, CustomResourceOptions? options = null)
+            : base("kubernetes:apps/v1:Deployment", name, SetAPIKindAndVersion(args), MakeResourceOptions(options, ""))
```

Comment:
It might not matter at this point, but for TS, Alex split the type string out into a separate variable for use with query features.

Reply:
Yeah - we didn't yet match that in the TF .NET SDK - so I've aligned with other .NET projections here for now.  We can follow up on this if/when needed and will need to do so for .NET as well.  Better not to implement now until we know how we'll want to use this across .NET.